mmv1/third_party/terraform/services/apphub/data_source_apphub_discovered_service.go

```diff
@@ -0,0 +1,185 @@
+package apphub
+
+import (
+    "fmt"
+    "time"
+
+    "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+    "github.com/hashicorp/terraform-provider-google/google/tpgresource"
+    transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceApphubDiscoveredService() *schema.Resource {
+    return &schema.Resource{
+        Read: dataSourceApphubDiscoveredServiceRead,
+        Schema: map[string]*schema.Schema{
+                        "location":{
+                            Type: schema.TypeString,
+                            Required: true,
+                        },
+                        "service_uri": {
+                            Type: schema.TypeString,
+                            Required: true,
+                        },
+                        "discovered_service": {
```

Comment:
Per a prior comment, I think we probably want this to be `discovered_services`.

Reply:
We are returning a single data source instance only. We are using list instead of map because map is not supported by recent terraform version (https://github.com/hashicorp/terraform-plugin-sdk/issues/616)